pkg/rancher-desktop/backend/images/imageProcessor.ts

```diff
@@ -154,27 +154,22 @@ export abstract class ImageProcessor extends EventEmitter {
 
   /**
    * Wrapper around the trivy command to scan the specified image.
-   * @param taggedImageName
+   * @param taggedImageName - fully specified image name, including registry and tag. nginx → docker.io/library/nginx:latest
+   * @param namespace - the containerd namespace of the image. Ignored when using moby.
    */
-  async scanImage(taggedImageName: string): Promise<childResultType> {
-    return await this.runTrivyCommand([
-      '--quiet',
-      'image',
-      '--format',
-      'json',
-      taggedImageName,
-    ]);
-  }
+  abstract scanImage(taggedImageName: string, namespace: string): Promise<childResultType>;
 
   /**
-   * Run trivy with the given arguments; the first argument is generally a
-   * subcommand to execute.
+   * Run trivy inside the VM with the given arguments; the 'trivy' command must be included
+   * in the args, so it can be called like `CONTAINERD_NAMESPACE=k8s.io trivy image ...`.
    */
-  async runTrivyCommand(args: string[], sendNotifications = true): Promise<childResultType> {
-    const subcommandName = args[0];
-    const child = this.executor?.spawn('trivy', ...args);
+  async runTrivyCommand(args: string[]): Promise<childResultType> {
+    // subcommandName is not used by processChildOutput() when args is defined
+    const subcommandName = 'image';
+    // must run as root to get access to the containerd socket
+    const child = this.executor?.spawn({ root: true }, ...args);
 
-    return await this.processChildOutput(child, subcommandName, sendNotifications, args);
+    return await this.processChildOutput(child, subcommandName, true, args);
```

Comment:
```suggestion
    return await this.processChildOutput(child, subcommandName, false, args);
```
Why the heck are we passing `true` here for `sendNotifications` only to add a hack to disable its effect when processing the argument‽

If you're doing this so we get `stderr` but not `stdout`, then we need separate flags for those.